src/aosm/azext_aosm/delete/delete.py

```diff
@@ -0,0 +1,330 @@
+        try:
+            poller = self.api_clients.aosm_client.network_service_design_versions.begin_delete(
+                resource_group_name=self.config.publisher_resource_group_name,
+                publisher_name=self.config.publisher_name,
+                network_service_design_group_name=self.config.nsd_name,
+                network_service_design_version_name=self.config.nsd_version,
+            )
+            LongRunningOperation(self.cli_ctx, "Deleting NSDV...")(poller)
+            print("Deleted NSDV.")
+        except Exception:
+            logger.error(
+                "Failed to delete NSDV %s from group %s",
+                self.config.nsd_version,
+                self.config.nsd_name,
+            )
+            raise
+
+    def delete_artifact_manifest(self, store_type: str) -> None:
+        """
+        _summary_
+
+        :param store_type: "sa" or "acr"
+        :raises CLIInternalError: If called with any other store type         :raises
+                Exception if delete throws an exception
+        """
+        if store_type == "sa":
+            assert isinstance(self.config, VNFConfiguration)
+            store_name = self.config.blob_artifact_store_name
+            manifest_names = [self.config.sa_manifest_name]
+        elif store_type == "acr":
+            store_name = self.config.acr_artifact_store_name
+            manifest_names = self.config.acr_manifest_names
+        else:
+            from azure.cli.core.azclierror import CLIInternalError
+
+            raise CLIInternalError(
+                "Delete artifact manifest called for invalid store type. Valid types"
+                " are sa and acr."
+            )
+
+        for manifest_name in manifest_names:
+            message = f"Delete Artifact manifest {manifest_name} from artifact store {store_name}"
+            logger.debug(message)
+            print(message)
+            try:
+                poller = self.api_clients.aosm_client.artifact_manifests.begin_delete(
+                    resource_group_name=self.config.publisher_resource_group_name,
+                    publisher_name=self.config.publisher_name,
+                    artifact_store_name=store_name,
+                    artifact_manifest_name=manifest_name,
+                )
+                LongRunningOperation(self.cli_ctx, "Deleting Artifact manifest...")(
+                    poller
+                )  # noqa: E501
+                print("Deleted Artifact Manifest")
+            except Exception:
+                logger.error(
+                    "Failed to delete Artifact manifest %s from artifact store %s",
+                    manifest_name,
+                    store_name,
+                )
+                raise
+
+    def delete_nsdg(self) -> None:
+        """Delete the NSD."""
+        assert isinstance(self.config, NSConfiguration)
+        message = f"Delete NSD {self.config.nsd_name}"
+        logger.debug(message)
+        print(message)
```

Comment:
Same as above. There are many similar places below, so I won't repeat the comments.

Reply:
We need to rethink how we output to console. These changes are too widespread to make before GA release, so I've opened work item https://dev.azure.com/msazuredev/AzureForOperators/_workitems/edit/992775 to track.